addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/EnvFileScanRule.java

```diff
@@ -38,8 +51,19 @@ public int getId() {
 
     @Override
     public boolean isFalsePositive(HttpMessage msg) {
+        HttpResponseHeader responseHeader = msg.getResponseHeader();
+        String contentType = responseHeader.getNormalisedContentTypeValue();
         String responseBody = msg.getResponseBody().toString();
-        // It likely is a FP if the response contains neither a comment nor assignment
-        return !responseBody.contains("#") && !responseBody.contains("=");
+
+        if (responseHeader.isEmpty() || !isPage200(msg) || responseBody.length() > RESPONSE_LEN_MAX)
+            return true;
+
+        if (contentType == null || contentType.equals("application/octet-stream")) {
+            boolean hasComments = COMMENT_PATTERN.matcher(responseBody).find();
+            boolean hasKeyvalue = KEYVAL_PATTERN.matcher(responseBody).find();
+            return !hasComments && !hasKeyvalue;
```

Comment:
It's missing coverage on one branch.

Reply:
I'm not sure what you are referring to here? I believe all the branches are covered in tests at the moment.

Reply:
Check JaCoCo report (I read it wrong, it's actually missing more).

Reply:
I believe in Eclipse you can also use the context menu (right click) "Coverage As > JUnit Test" then when you go back to the scan rule it should highlight and note what was/wasn't tested.

Reply:
I'm not sure 100% coverage is possible, at least on the test for isPage(200) path.
isFalsePositive is only called from https://github.com/zaproxy/zap-extensions/blob/master/addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/AbstractAppFilePlugin.java#L169-L178 when the status code is 200.
Shouldnt it be called in all cases, otherwise an rule cant mark a non 200 code as an FP?

Reply:
I'd have to go through the history to see why it was done like that. I "think" the logic was that it only makes sense to check content etc if the request is successful.

There are definitely still things that need to be adapted to Custom Pages (like the lines you linked).

Reply:
Just to be clear the coverage comment is just about this line not others.

Reply:
Ahh ha! I think I got it. Some additional regex needed

Reply:
As correctly pointed out earlier the status code check is somewhat redundant in this method, might be better to remove and revisit when the parent class is changed.